### ISSUE TITLE
Overriding a variable from a parent scope do not raise a warning

### ISSUE THREAD

**Issue by fungiboletus** [golangci/golangci-lint#653]
Hei, I have the following code and I would expect to get a warning or an error (I get none) :

```go
package main

import "fmt"

func main() {
	canard := ""
	if true {
		canard := "bonjour"
		fmt.Println(canard)
	}
	fmt.Println(canard)
}
```

1. Version of golangci-lint: `golangci-lint has version 1.17.1 built from 4ba2155`
2. Config : `golangci-lint run --enable-all`

**Comment by alexkohler** [comment]
The `shadow` flag is not enabled by default for vet. In order to get this behavior, you must explicitly enable it, i.e.
```
govet:
  # report about shadowed variables
  check-shadowing: true
```

**Comment by fungiboletus** [comment]
Alright. I guess it makes sense even though I would expect `--enable-all` to enable all options. I'm actually a bit surprised that `check-shadowing` isn't enabled by default. Thanks for the help.
